src/relayclient/GSNConfigurator.ts

```diff
@@ -1,42 +1,30 @@
-import { Address } from './types/Aliases'
+import { HttpProvider } from 'web3-core'
+import { Address, AsyncApprove, PingFilter, RelayFilter } from './types/Aliases'
 import { defaultEnvironment } from './types/Environments'
+import HttpClient from './HttpClient'
+import ContractInteractor from './ContractInteractor'
+import KnownRelaysManager, { EmptyFilter } from './KnownRelaysManager'
+import AccountManager from './AccountManager'
+import RelayedTransactionValidator from './RelayedTransactionValidator'
+import HttpWrapper from './HttpWrapper'
+import { EmptyApprove, GasPricePingFilter } from './RelayClient'
 
 const GAS_PRICE_PERCENT = 20
 const MAX_RELAY_NONCE_GAP = 3
 const DEFAULT_RELAY_TIMEOUT_GRACE_SEC = 1800
 
 const defaultGsnConfig: GSNConfig = {
-  contractInteractorConfig: {
-    gtxdatanonzero: defaultEnvironment.gtxdatanonzero,
-    verbose: false
-  },
-  relayClientConfig: {
-    gasPriceFactorPercent: GAS_PRICE_PERCENT,
-    minGasPrice: 0,
-    maxRelayNonceGap: MAX_RELAY_NONCE_GAP,
-    verbose: false,
-    relaySelectionManagerConfig: {
-      sliceSize: 3,
-      verbose: false
-    }
-  },
-  knownRelaysManagerConfig: {
-    relayTimeoutGrace: DEFAULT_RELAY_TIMEOUT_GRACE_SEC,
-    verbose: false
-  },
-  transactionValidatorConfig: {
-    verbose: false
-  },
-  relayProviderConfig: {
-    verbose: false
-  },
-  accountManagerConfig: {
-    verbose: false,
-    methodSuffix: '',
-    jsonStringifyRequest: false
-  },
+  gtxdatanonzero: defaultEnvironment.gtxdatanonzero,
+  gasPriceFactorPercent: GAS_PRICE_PERCENT,
+  minGasPrice: 0,
+  maxRelayNonceGap: MAX_RELAY_NONCE_GAP,
+  sliceSize: 3,
+  relayTimeoutGrace: DEFAULT_RELAY_TIMEOUT_GRACE_SEC,
+  methodSuffix: '',
+  jsonStringifyRequest: false,
   chainId: defaultEnvironment.chainId,
-  relayHubAddress: '0x0000000000000000000000000000000000000000'
+  relayHubAddress: '0x0000000000000000000000000000000000000000',
```

Comment:
why should we put a broken value for relayHub?

Reply:
What should I put? At least zero address is well-understood missing value. RelayClient is not functional without relay hub and I hate defaults, as you've probably noticed.

Reply:
it shoud be added with a type, but not with a value.

Reply:
there's no default for it. so leave it out.

Reply:
There are no nullable types in GSNConfig.